gmcs/constants.py

```diff
@@ -33,3 +33,26 @@
 OV_ORDERS = ['sov', 'ovs', 'osv', 'v-final']
 VFINAL = ['sov','osv','v-final']
 VO_ORDERS = ['svo', 'vos', 'vso', 'v-initial']
+
+### WH-QUESTIONS
+
+IN_SITU = 'in-situ' # No question phrase fronting
+EMB_INSITU = 'embed-insitu' # Embedded in situ questions (when fronting is also possible; that's rare)
```

Comment:
I think constants must be used for the sake of minimizing typos (of which we have lots and lots, and occasionally it costs hours of painful debugging).

As for what they should be -- that is of course another matter. The familiar tension between line length and explicitness...

Reply:
The comment helps, but I agree that constants (and variable names in general) should generally be more descriptive, and `EMB` frequently occurs in comments as Emily's initials so I was confused here, too. `EMBED_IN_SITU` is only 3 more characters.

Reply:
@olzama definitely agree constants are great. I just think these names could be a little longer (as @goodmami suggests).

Reply:
I renamed the EMB-one but note that we have lots of issues in the code with long lines and inconsistent indentation in the TDL as it is. Renaming constants should in fact also involve indentation/space/linebreak cleanup, which is a separate tasks. (And to continue hinting at something I think about this issue generally: I do not believe there is any hope for bringing a project like the Matrix to a consistent state, ever. Why? Because in order for sucn initiatives to lead, on average, to anything apart from lengthy discussions, a military-grade authority is needed, which I don't think will ever be present in the project, nor would any of us want it to be, I don't think. You might think right now: oh but it is Olga who is arguing over this, but trust me, if we try to institute any kind of protocol, you will soon find yourself arguing over something you didn't even expect you would.)

Reply:
One thing at a time :)

Regarding consistency, this is why PEP-8 and projects like [Black](https://github.com/psf/black) exist. Just defer to those rules and remove the question. But there's still a bit to do before we can get there.

Reply:
> this is why PEP-8 and projects like [Black](https://github.com/psf/black) exist

Oh, I know they exist but I still believe what I said. In order for a project to actually follow any guidelines in practice, a military-grade authority is needed within the project. Regardless of whether the guide came from. So our disagreement lies in you thinking we can get there and me thinking we cannot. I don't claim to actually know who is right of course :).

Reply:
I think a major piece of infrastructure that is missing right now (that we should probably make an issue for since it’s not there yet) is PR hooks and pre-commit hooks. The way these things are managed at my work very well across Python, C++, PHP, JavaScript, and more, is when you submit a PR, the server tells you if you’re breaking the rules. You *can* break the rules if you get a reviewer to accept, but there’s lots of signals about style problems, test failures, typing problems, etc. Yes, they aren’t 100%, but I think Black’s philosophy about no-config reformatting is great because it removes a category of disagreement. 

There’s of course a balance to find between writing styling rules and using them, but moving towards automated tooling and commit-blocking checks is exactly the “military-grade authority” that will allow an async project like this to maintain style over time.

Last note: I have seen people write linter rules to complain about invalid constant names, so it’s possible if we want to do it :-) (and there‘s probably something existing we could use)

Reply:
I don't seem to be able to get my "military-grade authority" point across. What I mean is, it can only come from a person who has actual power over the project. Such as the project owner. This project has always emphatically exercised a very different philosophy. 

People will always have different opinions on any particular style issue. There is nothing inherently right or wrong in any particular style guide. Subscribing to one is either a personal decision, in which case it does not make sense to bring that discussion into a collaborative project at all, or it is a project-wide decision, but that is only possible if one person who is actually the owner and has actual power over everyone else (as in: if you don't agree, I fire you) imposes the decision on everybody. Maybe something like this will happen in the Matrix one day but for now I don't see it.

Reply:
That said, I personally wouldn't mind a tool which I could **optionally** run to get an idea about the degree to which my code complies with some style guide. I would probably run it and would probably even follow (some of) its suggestions. I would be very against anything that actually prevents commits though, in the context of this project. (Not that I have that military-grade say, either!)

Reply:
I want to note that we already seem to be spending more time arguing over style issues (in a meta-way) than, say, discussing analyses or other contentful aspects of the work. I realize it may seem that I am the instigator but (i) actually, I am not :) I push back on these things but I never initiate them; and (ii) just wait till a project member comes on board who has strong views on style which are different from Mike's and T.J.'s Or even an issue surfaces on which Mike and T.J. happen to disagree. 

No no, I really am of the opinion that style should be kept firmly at bay and, come to think of it, not even come into CR (because it simply is too distracting).

Reply:
To conclude my above thoughts, I would be for encouraging using an automated tool which highlights style locally according to some guide (PEP or whatever) and for more or less banning style from code reviews.